patchwork-extensions-item/src/main/java/net/minecraftforge/common/ToolType.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Minecraft Forge, Patchwork Project
+ * Copyright (c) 2016-2020, 2019-2020
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation version 2.1
+ * of the License.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+
+package net.minecraftforge.common;
+
+import java.util.Map;
+import java.util.regex.Pattern;
+
+import com.google.common.collect.Maps;
+
+public final class ToolType {
```

Comment:
This should be moved to a shared module so that block / blockstate extensions can access it too. It might be silly having a single-class module, but I think that's a cleaner solution than eg. making extensions-block depend on extensions-item.

Reply:
Not 100% sure I got everything for creating the new module, but it's been added.